cryostatio/cryostat
# fix(discovery): should send notifications on credential store update

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes:  https://github.com/cryostatio/cryostat-web/issues/805
Depends on https://github.com/cryostatio/cryostat-core/pull/172

## Description of the change:

- [x] Fixed Discovery Storage to always send WS notifications when new Jvmids are successfully computed.
- [x] Updated tests to check if notifications are sent (also apply fixes to check nested nested Environement node).

## Motivation for the change:

Front-end should be able to display newly computed jvmIds.

## How to manually test:
1. `Run CRYOSTAT_IMAGE=quay.io... sh smoketest.sh...`
2.  Choose any target with authentication (and no backend-saved credentials). See jvmId is null.
3. Add credentials.
4. See the updated jvmIds.
